src/main/java/core/basesyntax/service/converter/DataConverterImpl.java

```
@@ -0,0 +1,33 @@
+package core.basesyntax.service.converter;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertFruitTransactions(List<String> listTransactionFromFile) {
+        List<FruitTransaction> fruitTransactions = new ArrayList<>();
+
+        for (String string : listTransactionFromFile) {
+            String[] split = string.split(",");
+
+            FruitTransaction.Operation operation = convertOperation(split[0]);
+
+            FruitTransaction fruitTransaction = new FruitTransaction(
+                    operation, split[1], Integer.parseInt(split[2]));
+            fruitTransactions.add(fruitTransaction);
+        }
+
+        return fruitTransactions;
+    }
+
+    private FruitTransaction.Operation convertOperation(String operation) {
+        return switch (operation) {
+            case "b" -> FruitTransaction.Operation.BALANCE;
+            case "s" -> FruitTransaction.Operation.SUPPLY;
+            case "p" -> FruitTransaction.Operation.PURCHASE;
+            default -> FruitTransaction.Operation.RETURN;
```

Using a `default` case to handle all unknown operations by treating them as a `RETURN` can hide errors in the input data. The requirements state that the program should handle an 'incorrect strategy'. It would be more robust to throw an `IllegalArgumentException` or a custom `RuntimeException` here to signal that the input file contains an invalid operation code.